src/dayToDF.py

```diff
@@ -61,14 +63,11 @@ def gaps(df):
         output: list [int] of x values that are not in any words
         """
         # know edge of left most word is not a gap, so skip it
-        gaps = list(range(int(df['xmin'].min() + 1), int(df['xmax'].max())))
+        #gaps = list(range(int(df['xmin'].min() + 1), int(df['xmax'].max())))
 
         # remove each column of pixels where text exists
-        for _, word in df.iterrows():
-            for i in gaps[:]:
-                if word['xmin'] <= i <= word['xmax']:
-                    gaps.remove(i)
-                    continue
+        gaps = (i for i in range(int(df['xmin'].min() + 1), int(df['xmax'].max())) if not df.loc[(df['xmin'] <= i) & (df['xmax'] >= i)].empty)
```

Comment:
fancy!  with the list() typecaste commented out, is it still returning a list or a long tuple? And does that affect the behaviour downstream?

Reply:
No, gaps is just used to get an iterator, that is used once